server/service/src/invoice/inventory_adjustment/generate.rs

```diff
@@ -0,0 +1,120 @@
+use chrono::Utc;
+
+use repository::{InvoiceLineRow, InvoiceLineRowType, NameRowRepository, StockLine, StockLineRow};
+use repository::{
+    InvoiceRow, InvoiceRowStatus, InvoiceRowType, NumberRowType, RepositoryError, StorageConnection,
+};
+use util::constants::INVENTORY_ADJUSTMENT_NAME_CODE;
+use util::uuid::uuid;
+
+use crate::number::next_number;
+
+use super::{AdjustmentType, InsertInventoryAdjustment};
+
+pub fn generate(
+    connection: &StorageConnection,
+    store_id: &str,
+    user_id: &str,
+    InsertInventoryAdjustment {
+        stock_line_id,
+        adjustment,
+        adjustment_type,
+        inventory_adjustment_reason_id,
+    }: InsertInventoryAdjustment,
+    stock_line: StockLine,
+) -> Result<(InvoiceRow, InvoiceLineRow, StockLineRow), RepositoryError> {
+    let current_datetime = Utc::now().naive_utc();
+
+    let inventory_adjustment_name = NameRowRepository::new(connection)
+        .find_one_by_code(INVENTORY_ADJUSTMENT_NAME_CODE)?
+        .ok_or(RepositoryError::NotFound)?;
+
+    let invoice_number = next_number(
+        connection,
+        &match adjustment_type {
+            AdjustmentType::Addition => NumberRowType::InventoryAddition,
+            AdjustmentType::Reduction => NumberRowType::InventoryReduction,
+        },
+        store_id,
+    )?;
+
+    let invoice = InvoiceRow {
+        id: uuid(),
+        user_id: Some(user_id.to_string()),
+        name_link_id: inventory_adjustment_name.id,
+        r#type: InvoiceRowType::OutboundReturn,
```

Comment:
Whoops missed this one, should be InventoryAddition or InventoryReduction!

Reply:
Eek -- okay, done: ac2223464